Features/ManageabilityPkg/Library/ManageabilityTransportSerialLib/Dxe/ManageabilityTransportSerial.c

```diff
@@ -0,0 +1,377 @@
+                                        assigned to Serial transport interface.
+
+  @retval      EFI_SUCCESS              Transport interface is initialized
+                                        successfully.
+  @retval      EFI_INVALID_PARAMETER    The invalid transport token.
+  @retval      EFI_NOT_READY            The transport interface works fine but
+  @retval                               is not ready.
+  @retval      EFI_DEVICE_ERROR         The transport interface has problems.
+  @retval      EFI_ALREADY_STARTED      Teh protocol interface has already initialized.
+  @retval      Otherwise                Other errors.
+
+**/
+EFI_STATUS
+EFIAPI
+SerialTransportInit (
+  IN  MANAGEABILITY_TRANSPORT_TOKEN                 *TransportToken,
+  IN  MANAGEABILITY_TRANSPORT_HARDWARE_INFORMATION  HardwareInfo OPTIONAL
+  )
+{
+  CHAR16      *ManageabilityProtocolName;
+  EFI_STATUS  Status;
+
+  if (TransportToken == NULL) {
+    DEBUG ((DEBUG_ERROR, "%a: Invalid transport token.\n", __func__));
+    return EFI_INVALID_PARAMETER;
+  }
+
+  if (HardwareInfo.Serial == NULL) {
+    DEBUG ((DEBUG_MANAGEABILITY_INFO, "%a: Hardware information is not provided, use dfault settings.\n", __func__));
+    mSerialHardwareInfo.IpmiRequesterAddress = FixedPcdGet8 (PcdIpmiSerialRequesterAddress);
+    mSerialHardwareInfo.IpmiResponderAddress = FixedPcdGet8 (PcdIpmiSerialResponderAddress);
+    mSerialHardwareInfo.IpmiRequesterLUN     = FixedPcdGet8 (PcdIpmiSerialRequesterLUN);
+    mSerialHardwareInfo.IpmiResponderLUN     = FixedPcdGet8 (PcdIpmiSerialResponderLUN);
+  } else {
+    mSerialHardwareInfo.IpmiRequesterAddress = ((MANAGEABILITY_TRANSPORT_SERIAL_HARDWARE_INFO *)HardwareInfo.Serial)->IpmiRequesterAddress;
+    mSerialHardwareInfo.IpmiResponderAddress = ((MANAGEABILITY_TRANSPORT_SERIAL_HARDWARE_INFO *)HardwareInfo.Serial)->IpmiResponderAddress;
+    mSerialHardwareInfo.IpmiRequesterLUN     = ((MANAGEABILITY_TRANSPORT_SERIAL_HARDWARE_INFO *)HardwareInfo.Serial)->IpmiRequesterLUN;
+    mSerialHardwareInfo.IpmiResponderLUN     = ((MANAGEABILITY_TRANSPORT_SERIAL_HARDWARE_INFO *)HardwareInfo.Serial)->IpmiResponderLUN;
+  }
+
+  // Get protocol specification name.
+  ManageabilityProtocolName = HelperManageabilitySpecName (TransportToken->ManageabilityProtocolSpecification);
+
+  DEBUG ((DEBUG_MANAGEABILITY_INFO, "%a: Serial transport hardware for %s is:\n", __func__, ManageabilityProtocolName));
+  DEBUG ((DEBUG_MANAGEABILITY_INFO, "IPMI requester address       : 0x%08x\n", mSerialHardwareInfo.IpmiRequesterAddress));
+  DEBUG ((DEBUG_MANAGEABILITY_INFO, "IPMI responder address       : 0x%08x\n", mSerialHardwareInfo.IpmiResponderAddress));
+  DEBUG ((DEBUG_MANAGEABILITY_INFO, "IPMI requester LUN           : 0x%08x\n", mSerialHardwareInfo.IpmiRequesterLUN));
+  DEBUG ((DEBUG_MANAGEABILITY_INFO, "IPMI responder LUN           : 0x%08x\n", mSerialHardwareInfo.IpmiResponderLUN));
+
+  Status = SerialPortInitialize ();

```

Comment:
you can move this line to the begining of the function, if it fails no need to the initialization of mSerialHardwareInfo.